machine/accton/accton_as4510_52t/u-boot/platform-as4510_52t.patch

```diff
@@ -0,0 +1,2918 @@
++	.u_boot_cmd : { *(.u_boot_cmd) }
++	__u_boot_cmd_end = .;
++
++	. = ALIGN(4);
++
++	__image_copy_end = .;
++
++	.rel.dyn : {
++		__rel_dyn_start = .;
++		*(.rel*)
++		__rel_dyn_end = .;
++	}
++
++	.dynsym : {
++		__dynsym_start = .;
++		*(.dynsym)
++	}
++
++	.bss __rel_dyn_start (OVERLAY) : {
++		__bss_start = .;
++		*(.bss)
++		 . = ALIGN(4);
++		__bss_end__ = .;
++        _end = .;
++	}
++
++	.irqstack : {
++		__irqstack_end = .;
++		. = . + 0x2008;
++		__irqstack_start = .;
++	}
++
++	/DISCARD/ : { *(.dynstr*) }
++	/DISCARD/ : { *(.dynamic*) }
++	/DISCARD/ : { *(.plt*) }
++	/DISCARD/ : { *(.interp*) }
++	/DISCARD/ : { *(.gnu*) }
++}
+diff --git a/boards.cfg b/boards.cfg
+index 368f3c4..281aa26 100644
+--- a/boards.cfg
++++ b/boards.cfg
+@@ -308,6 +308,7 @@ hurricane2_nand              arm         armv7       bcm95615x           broadco
+ hurricane2_nor              arm         armv7        bcm95615x           broadcom       hurricane2        hurricane2:L2C_AS_RAM,RUN_DDR_SHMOO2,CUSTOM_LINKER_SCRIPT,NOR_IPROC_BOOT
+ hurricane2_noshmoo           arm         armv7       bcm95615x           broadcom       hurricane2        hurricane2:CUSTOM_LINKER_SCRIPT
+ hurricane2_rom_emul          arm         armv7       bcm95333x           broadcom       hurricane2        hurricane2_rom_emul:L2C_AS_RAM,NO_CODE_RELOC
++AS4510_52T                   arm         armv7       as4510_52t           accton        hurricane2        AS4510_52T:ONIE_PLATFORM_REV=0
+ katana2                       arm         armv7       bcm95645x           broadcom       katana2           katana2:RUN_DDR_SHMOO2,IPROC_DDR_ECC 
+ katana2_nand                  arm         armv7       bcm95645x           broadcom       katana2           katana2:RUN_DDR_SHMOO2,IPROC_DDR_ECC,NAND_IPROC_BOOT
+ origen			     arm	 armv7	     origen		 samsung	exynos
+diff --git a/broadcom_boards.mk b/broadcom_boards.mk
+index 60b2ecf..39959e1 100644
+--- a/broadcom_boards.mk
++++ b/broadcom_boards.mk
+@@ -151,6 +151,10 @@ hurricane2_rom_emul_config: $(obj)hurricane2_rom_emul_config
+ $(obj)hurricane2_rom_emul_config: $(TOPDIR)/include/configs/hurricane2_rom_emul.h 
+ 	@$(MKCONFIG) -A $(@F:_config=); touch $@
+ 
++AS4510_52T_config: $(obj)AS4510_52T_config
++$(obj)AS4510_52T_config: $(TOPDIR)/include/configs/AS4510_52T.h
++	@$(MKCONFIG) -A $(@F:_config=); touch $@
++
+ katana2_config: $(obj)katana2_config 
+ $(obj)katana2_config: $(TOPDIR)/include/configs/katana2.h  
+ 	@$(MKCONFIG) -A $(@F:_config=); touch $@
+diff --git a/common/cmd_mtdparts.c b/common/cmd_mtdparts.c
+index 2268829..d3abe91 100644
+--- a/common/cmd_mtdparts.c
++++ b/common/cmd_mtdparts.c
+@@ -174,7 +174,7 @@ static int device_del(struct mtd_device *dev);
+  * @param retptr output pointer to next char after parse completes (output)
+  * @return resulting unsigned int
+  */
+-static unsigned long memsize_parse (const char *const ptr, const char **retptr)
++static u64 memsize_parse (const char *const ptr, const char **retptr)
```

Comment:
This change for mtd size moving from u32 to u64 looks like a generally good thing.  It does not look machine specific.

I am guessing this is needed to handle NAND parts that are larger than 4GB.

I think it would be OK to promote this change as a general u-boot patch, i.e. put it in patches/u-boot/2012.10/.